File: Py_Projects/Euler/2.py
def Euler_2(top=4000000):
	'''Find the sum of all even-valued terms in the Fibonacci sequence
	that do not exceed the input
	'''
	total = 0
	(a, b) = (1, 1)
	while b <= top:
		(a, b) = (b, a+b)
		if b % 2 == 0 and b <= top:
			total += b
	return total


if __name__ == '__main__':
	print(Euler_2())

File: C_Projects/Euler/src/2.c
#include "Euler.h"

int64_t Euler_2()
{
    int sum = 0;
    int a = 1;
    int b = 2;
    while (b < 4000000) {
        if (b % 2 == 0) {
            sum += b;
        }
        int tmp = a;
        a = b;
        b = tmp + b;
    }
    return (int64_t) sum;
}
